### ISSUE TITLE
Add CoAtNet

### ISSUE THREAD

**Issue by IMvision12** [keras-team/keras-cv#1223]
**Short Description**

Contnet is a network which is combination of depthwise Convolution and self-Attention, stacking convolution layers and attention layers vertically. CoAtNet achieves 86.0% ImageNet top-1 accuracy; When pre-trained with 13M images from ImageNet-21K, CoAtNet achieves 88.56% top-1 accuracy, 

Accuracy Graph :
![Capture](https://user-images.githubusercontent.com/88665786/210107373-eda24426-148c-4c37-ae9b-aac96eb2809b.PNG)

[Paper](https://arxiv.org/abs/2106.04803)

@tanzhenyu Will it be worthwhile to include this model? if yes I Would like to work on this!!!

**Comment by tanzhenyu** [comment]
@IMvision12 there are a couple of models open for contribution, search for label "contribution welcome", for example, swin transformer.
For CoAtNet, we need to verify the impact -- as a backbone, what will this be used for?
